src/decompress.cpp

```diff
@@ -29,4 +30,54 @@ auto read_header(huffman::bit_span& compressed_bits)
   compressed_bits.consume(3);
   return BlockHeader{final, type};
 }
-}  // namespace starflate::detail
+
+}  // namespace detail
+
+auto decompress(std::span<const std::byte> src, std::span<std::byte> dst)
+    -> std::expected<DecompressResult, DecompressError>
+{
+  using enum detail::BlockType;
+
+  huffman::bit_span src_bits{src};
+  while (true) {
+    const auto header = detail::read_header(src_bits);
+    if (not header) {
+      return std::unexpected{header.error()};
+    }
+    if (header->type == NoCompression) {  // no compression
+      // Any bits of input up to the next byte boundary are ignored.
+      src_bits.consume_to_byte_boundary();
+      const std::uint16_t len = src_bits.pop_16();
```

Comment:
Do we know the input is large enough to store this?

Reply:
Guessing you mean "Do we know `src_bits.size() >= 16`"?

`bit_span.pop()` asserts that, but we should probably handle it here, in the same way that we handle the "not enough bits in src" issue below. I'll do that in another PR, this one is getting a little big.

Reply:
yeah exactly